### PR TITLE
FM2-589: Add support for Json patching operations - Encounter Resource

### PR DESCRIPTION
**Pull request by mherman22**
## Description of what I changed

Ensure Encounter resource(R4) supports json merge patch operations.

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-589

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java [hunk 1]
```diff
@@ -106,6 +114,62 @@ public void shouldReturnExistingEncounterAsJson() throws Exception {
 		assertThat(encounter, validResource());
 	}
 	
+	@Test
+	public void shouldPatchExistingMedicationViaJsonMergePatch() throws Exception {
```
Comment (ibacher) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java:118]: Check the naming of functions. These should also be immediately after the `update` tests.

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java [hunk 2]
```diff
@@ -49,8 +53,12 @@ public class EncounterFhirResourceProviderIntegrationTest extends BaseFhirR4Inte
 	
 	private static final String MEDICATION_REQUEST_QUERY_INITIAL_DATA_XML = "org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImplTest_initial_data.xml"; // not loaded for all tests
 	
+	private static final String JSON_PATCH_ENCOUNTER_PATH = "org/openmrs/module/fhir2/providers/Encounter_json_patch.json";
```
Comment (ibacher) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java:56]: Could you rearrange these so it goes:

1. Create
1a. Create JSON
1b. Create XML
2. Patch
2a. Patch JSON

Thanks

Comment (mherman22) [reply integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java:58]: Done

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java [hunk 3]
```diff
@@ -814,6 +822,62 @@ public void shouldThrow404WhenUpdatingNonExistingVisitAsJson() throws Exception
 		assertThat(response.getContentAsString(), notNullValue());
 	}
 	
+	@Test
+	public void shouldPatchExistingMedicationUsingJsonMergePatch() throws Exception {
```
Comment (ibacher) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java:826]: @mherman22 I missed this when reviewing. Could you submit a new PR renaming these methods?

Comment (mherman22) [reply integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java:826]: changes are part of https://github.com/openmrs/openmrs-module-fhir2/pull/506

**Pull request merged**